### ISSUE TITLE
xwayland transparent areas are filled with garbage pixels

### ISSUE THREAD

**Issue by YaLTeR** [swaywm/sway#3395]
* Sway Version: `1.0-beta.2-136-g14cab786 (Jan  9 2019, branch 'master')`
* Debug Log: <https://gist.github.com/YaLTeR/b2912d599d8a0e79657541fd54b1d1cb>
* Configuration File: default

Transparent or otherwise invalid areas on xwayland windows are filled with garbage pixels. They seem to contain uncleaned previous data from the framebuffer or something:
![image](https://user-images.githubusercontent.com/1794388/50927795-e81f5580-1469-11e9-8329-6a60230607fc.png)

Easiest to see on  `xeyes`, but it's also possible to see garbage pixels on other xwayland clients (like firefox, chromium) when expanding their size.

Arch Linux x64, AMD RX 580, AMDGPU.

**Comment by emersion** [comment]
Yeah, this is a damage tracking issue. I believe it's also reproducible on rootston (will be easier to debug there).

**Comment by emersion** [comment]
Disclaimer: not sure we can do anything about this.

**Comment by moretheard** [comment]
amdgpu - kernel 4.20.1
sway 1.0-beta.2 
kitty 0.13.2

wayland and x backend are affected.

Another way to reproduce: resize: $mod+r and blank areas are filled with dead pixels.

And another one same issue?

If i start a term (kitty) blank term. I have to start another instance and switch between them, then both show up.

**Comment by Nikki1993** [comment]
I think I am experiencing the same issue on my RX 580 but I noticed that in my case it could also be nonblank areas. The best way to reproduce that one was to install the Nyan Cat loading bar extension for Webstorm (or any similar Intelij product) and then load a large project. You will notice that the rainbow loading bar is all messed up while on Intel GPU it's all fine and dandy. 

But for the majority of instances, it's the same as for the rest, blank areas = garbage pixels.

**Comment by Emantor** [comment]
AFAICS this has been fixed on Xwayland master, using `xeyes` with XWayland 1.20.8 and sway master, The transparent area is now black. Tested on a RX580.

**Comment by emersion** [comment]
Have you tried resizing xeyes?

**Comment by Emantor** [comment]
Hm, I see presumed aliasing artifacts which produce a white halo around the eyes, but the background remains black correctly.

**Comment by Limero** [comment]
I can confirm that the background is black but there are artifacts when resizing on my RX480.
![image](https://user-images.githubusercontent.com/639787/79632772-aa77a500-8161-11ea-990e-6ed2360c8a2c.png)

**Comment by aplund** [comment]
Isn't this more of an issue with support for non-rectangular window shapes?  The "invalid" area should be masked out.  I don't understand how xwayland works with this protocol.

**Comment by Geo25rey** [comment]
There is still flickering and artifactting when resizing the `xeyes` window.

**Comment by aplund** [comment]
This gist has code for a basic X client that shows the transparent region as black.

https://gist.github.com/aplund/8c7465aa2619a7f3ebb306224377e1c7

Running the same client on Xwayland on weston gives the correct results.

**Comment by aqxa1** [comment]
> This gist has code for a basic X client that shows the transparent region as black.
> 
> https://gist.github.com/aplund/8c7465aa2619a7f3ebb306224377e1c7
> 
> Running the same client on Xwayland on weston gives the correct results.

I haven't tested with Weston but the test program above still renders the transparent background as black with git Xwayland/sway/wlroots.
